Fix shellcheck violations

theneva opened the pull request:
- Stopped using `local`
- Ditched reassigning `$SECRETS_DIR` to `$secrets_dir` (we already check that it's set)
- Double quotes around all variable references
- All string replacement is now done with `tr`

SimenB commented on .gitignore:
```diff
@@ -1,3 +1,6 @@
 
 build/
+
+*.swo
```
fix your global gitignore

theneva replied:
no

SimenB replied:
or use a proper editor

SimenB replied:
or fix your vim to put those file somewhere not CWD

SimenB commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
-  local message=$1
-  local iso_8601_timestamp=$(date +%Y-%m-%dT%H:%M:%S%z)
-  echo '{"@version": 1, "@timestamp": "'"$iso_8601_timestamp"'", "level": "INFO", "message": "'"$1"'"}'
+  message=$1
```
shellscript newbie, but are variables automatically scoped to their function?

theneva replied:
nope, globals all the way. I think there's no way to avoid this in `sh` if I can't use `local`, except running everything as different processes

tfheen replied:
You can use subshells, but ugh. `local` is fine as long as you don't care about Solaris and other OS-es where /bin/sh isn't POSIX sh.

tfheen replied:
Also, quoting.

theneva replied:
/bin/sh points to busybox here, which supports local. Should I just use that and ignore Shellcheck?

SimenB replied:
Yeah, and use https://github.com/koalaman/shellcheck/wiki/Directive if possible to disable just the `local` warning?

tfheen replied:
Or follow the advice on https://github.com/koalaman/shellcheck/wiki/SC2155 ? I assume that's what it's warning about, not some other error?

theneva replied:
It's warning about https://github.com/koalaman/shellcheck/wiki/SC2039:

For example, adding `local` in front of `message="$1"` on line 4:

```
In scripts/run-app.sh line 4:
  local message="$1"
  ^-- SC2039: In POSIX sh, 'local' is undefined.
```

tfheen replied:
That'd I'd just silence, yeah.

SimenB commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
-  local secrets_dir=$1
+  for filename in "$SECRETS_DIR"/*; do
+    filename_without_path=${filename##*/}
+    filename_uppercased=$(echo "$filename_without_path" | tr '[:lower:]' '[:upper:]');
```
fancy `tr`

theneva replied:
it yelled at me because the old one didn't support accented characters etc

tfheen replied:
add quoting?

SimenB commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
   # npm provides binaries on PATH in scripts, so we have to do the same
-  export PATH=$PATH:$(pwd)/node_modules/.bin
+  path_with_node_modules_binaries=$PATH:$(pwd)/node_modules/.bin
+  export PATH=$path_with_node_modules_binaries
```
we don't have to export it, it's enough to just prepend it to the `node` command. that way we don't have to mutate environment variables

theneva replied:
that's better!

SimenB commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
-  if [ -z ${FIAAS_ENVIRONMENT} ]; then
-    log "FIAAS_ENVIRONMENT is unset, not looking for secrets";
+  if [ -z "$FIAAS_ENVIRONMENT" ]; then
+    log "FIAAS_ENVIRONMENT is unset; not looking for secrets";
```
heh

theneva replied:
that was a Martincheck violation, not shellcheck

SimenB commented:
You've tested this? Both with and without secrets with, with and without content?

theneva commented:
yes

tfheen commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
 export_secrets_from_dir() {
-  local secrets_dir=$1
+  for filename in "$SECRETS_DIR"/*; do
+    filename_without_path=${filename##*/}
```
quoting?

tfheen commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
-  echo '{"@version": 1, "@timestamp": "'"$iso_8601_timestamp"'", "level": "INFO", "message": "'"$1"'"}'
+  message=$1
+  iso_8601_timestamp=$(date +%Y-%m-%dT%H:%M:%S%z)
+  echo '{"@version": 1, "@timestamp": "'"$iso_8601_timestamp"'", "level": "INFO", "message": "'"$message"'"}'
```
What ensures message does not contain `"`? You might also want to use printf to avoid all the quoting.

theneva replied:
That… is very true. Good catch! Now to figure out how to escape the `"`s…

tfheen commented on scripts/run-app.sh:
```diff
@@ -1,55 +1,56 @@
     log "Exporting secret '$filename_without_path' as '$secret_name'"
 
-    local secret=$(cat $filename);
+    secret=$(cat "$filename");
```
Quoting of `$()` too.

theneva replied:
right… thanks 😄

theneva commented:
@tfheen I think I've fixed it all now. Also changed logging to account for messages with `"` by `sed`-ing them to `\"`. Mind having one final look before I merge?

theneva commented:
@SimenB does this look good to you? I've tested it again with all permutations I could think of, and it all seems to work
